libs/core/src/main/java/org/opensearch/core/common/io/stream/BaseStreamOutput.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.core.common.io.stream;
+
+import java.io.OutputStream;
+
+/**
+ * Foundation class for writing core types over the transport stream
+ *
+ * todo: refactor {@code StreamOutput} primitive writers to this class
+ *
+ * @opensearch.internal
+ */
+public abstract class BaseStreamOutput extends OutputStream {}
```

Comment:
Similar to another PR where we're doing quite a bit of `Base*`, should this have been called `OutputStream extends java.io.OutputStream`, or at least `BaseOutputStream extends ...` to match the order of StreamOutput and OutputStream?